packages/firestore/test/unit/specs/write_spec.test.ts

```diff
@@ -637,4 +637,74 @@ describeSpec('Writes:', [], () => {
       .writeAcks(1, { expectUserCallback: false })
       .expectNumOutstandingWrites(0);
   });
+
+  specTest(
+    'Pending writes are shared between clients',
+    ['multi-client'],
+    () => {
+      const query = Query.atPath(path('collection'));
+      const docV1 = doc(
+        'collection/a',
+        0,
+        { v: 1 },
+        { hasLocalMutations: true }
+      );
+      const docV2 = doc(
+        'collection/a',
+        0,
+        { v: 2 },
+        { hasLocalMutations: true }
+      );
+      const docV3 = doc(
+        'collection/a',
+        0,
+        { v: 3 },
+        { hasLocalMutations: true }
+      );
+
+      return client(0)
+        .userListens(query)
+        .watchAcksFull(query, 500)
+        .expectEvents(query, {})
+        .userSets('collection/a', { v: 1 })
+        .expectEvents(query, {
+          hasPendingWrites: true,
+          added: [docV1]
+        })
+        .client(1)
+        .userListens(query)
+        .expectEvents(query, {
+          hasPendingWrites: true,
+          // TODO(multitab): `fromCache` should reflect the primary client's
+          // networking state (and be false).
```

Comment:
I'm glad you flagged this.  Do you have a plan to fix it?  I think fromCache=false means the primary tab has received a CURRENT message for the particular query since the last time the tab was in OnlineState.Offline.  Do we need to persist this (on a per-query basis) in IndexedDB or WebStorage somehow?